prow/cmd/deck/main.go

```diff
@@ -1222,16 +1244,41 @@ func handleProwJob(prowJobClient prowv1.ProwJobInterface) http.HandlerFunc {
 }
 
 // canTriggerJob determines whether the given user can trigger any job.
-func canTriggerJob(user string, cfg *config.RerunAuthConfig) bool {
+func canTriggerJob(user string, pj prowapi.ProwJob, cfg *config.RerunAuthConfig, cli prowgithub.Client) (bool, error) {
 	if cfg.AllowAnyone {
-		return true
+		return true, nil
 	}
 	for _, u := range cfg.AuthorizedUsers {
 		if prowgithub.NormLogin(u) == prowgithub.NormLogin(user) {
-			return true
+			return true, nil
 		}
 	}
-	return false
+	jobPermissions := pj.Spec.RerunPermissions
+	if jobPermissions == nil {
```

Comment:
Why did we keep this as a pointer?

Reply:
Discussed on Slack -- kept it as a pointer in config, made it a struct in CRD